apps/spruce/src/components/Table/BaseTable.tsx

```diff
@@ -123,9 +129,12 @@ export const BaseTable = forwardRef(
                         ) : (
                           <TableSearchPopover
                             data-cy={meta?.search?.["data-cy"]}
-                            onConfirm={(value) =>
-                              header.column.setFilterValue(value)
-                            }
+                            onConfirm={(value) => {
+                              header.column.setFilterValue(value);
+                              if (table) {
+                                table.firstPage();
+                              }
```

Comment:
I think it would be:
```suggestion
                              if (usePagination) {
                                table?.firstPage();
                              }
```
Could you apply this to the `TableFilterPopover` as well?